feat(general): amended all lambda runtime deprecations thru Jun 30 2026 (updated 12/15/25 for Python 3.9)

RoryKiefer opened the pull request:
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

# Updating Deprecation Information in Code Commentary


## Description

* [Data source](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) for deprecation timeline
* Updating cloudformation and terraform tests that were using python3.9 under circumstances where the runtime version was not meant to be the failure from 3.8/3.9 -> 3.13

Fixes # [7283](https://github.com/bridgecrewio/checkov/issues/7283)

RoryKiefer committed:
amended all lambda runtime deprecations thru Jun 30 2026

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

maxamel committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

maxamel committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

maxamel committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
appeasing linter

RoryKiefer committed:
Merge branch 'deprecated-lambdas-thru-06-30-26' of github.com:RoryKiefer/checkov into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

maxamel committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
fixing unit tests

RoryKiefer committed:
fixing cfn-lint

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

maxamel commented:
hi @RoryKiefer , in the recent commit I found some csv files that seem unnecessary. Can you elaborate why they are needed in this PR ?

RoryKiefer committed:
reverting python to nodejs for test check

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Delete 20251117-111144_container_images.csv

RoryKiefer committed:
Delete 20251117-111144_iac.csv

RoryKiefer committed:
Delete console

RoryKiefer committed:
Delete 20251117-111144_oss_packages.csv

RoryKiefer commented:
@maxamel python 3.9 runtime is deprecated today. Firing up a code change!

RoryKiefer committed:
updates: tests using python3.9 lambda runtime updated to use 3.13 instead; added python3.9 to deprecated runtime datastructure

RoryKiefer committed:
Delete 20251215-110839_oss_packages.csv

RoryKiefer committed:
Delete 20251215-110839_iac.csv

RoryKiefer committed:
Delete 20251215-110839_container_images.csv

RoryKiefer commented:
> > @maxamel not sure where those CSV files came from -- perhaps an artifact of the tests I ran? Either way, they've been cleaned up.
> 
> 
> 
> @RoryKiefer I see they are still there. Can you re-check?

@maxamel done. I'll check back in a day or two and make sure they're still gone? It looks like when I ran the test suite on 12/15/25 the CSV files came back. Perhaps I should add these to the repos ignore file.

RoryKiefer committed:
ignoring test output csv files

RoryKiefer committed:
Merge branch 'deprecated-lambdas-thru-06-30-26' of github.com:RoryKiefer/checkov into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
fixed tests

RoryKiefer commented:
@maxamel merging main into my branch introduced a unit test failure. I fixed that failure, though, so I fixed the error in main as a result. It had to do with the pass/fail versions of AWS transfer server.

Also, I added those csv files to .gitignore so there will be no more unwanted csv files.

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Update exampleSAM.yaml

conforming to 3.13 as "testing-pass" runtime

RoryKiefer committed:
Update example.yaml

keeping with nodejs instead of overusing python for runtime test content

RoryKiefer committed:
Update exampleSAM.yaml

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26

RoryKiefer committed:
Merge branch 'main' into deprecated-lambdas-thru-06-30-26
